enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberStateMachineTest.java

```diff
@@ -463,6 +504,171 @@ public Object answer( InvocationOnMock invocation ) throws Throwable
 
     }
 
+    @Test
+    public void whenHAModeSwitcherSwitchesToSlaveTheOtherModeSwitcherDoNotGetTheOldMasterClient() throws Throwable
+    {
+        InstanceId me = new InstanceId( 1 );
+        StoreId storeId = new StoreId();
+        HighAvailabilityMemberContext context = mock( HighAvailabilityMemberContext.class );
```

Comment:
well exactly, to have something readable and in best case reusable, instead of this :sos:

was thinking about extracting clusterMember, update puller and dependency resolver mocks,
but if you think its too much for it,  that's fair enough